sklearn/cluster/_kmeans.py

```diff
@@ -206,7 +208,7 @@ def _kmeans_plusplus(X, n_clusters, x_squared_norms, random_state, n_local_trial
         n_local_trials = 2 + int(np.log(n_clusters))
 
     # Pick first center randomly and track index of point
-    center_id = random_state.randint(n_samples)
+    center_id = random_state.choice(n_samples, replace=False, p=sample_weight)
```

Comment:
@glemaitre this is a breaking change even when sample_weight=None (transformed into ones at this point). Do you find it acceptable ?

Reply:
I think this is fine. Only the randomness is different but the behaviour is the same. We can document this change in the "Change models". Since we add support for `sample_weight` I think that we can go ahead here.